### PR TITLE
Move Azure content into dotnet/docs repo

### PR DESCRIPTION
**Pull request by CamSoper**
Staging link:  https://review.docs.microsoft.com/en-us/dotnet/azure/?branch=pr-en-us-17473

## Summary

Relocating conceptual content from Azure/azure-docs-sdk-dotnet.

This has the following intended effects:

1. Better, more manageable write/build/publish loop. Azure/azure-docs-sdk-dotnet is primarily for the reference content and has a bunch of automated processes attached to it that complicate writing and publishing.
1. Takes advantage of the processes and tools already in this repo.
1. Being with the .NET docs should hopefully encourage more community contributions.

### What was done

1. Created an *docs\azure* folder. This will serve requests to https://docs.microsoft.com/dotnet/azure by default. Since the .NET docs are higher in the OPS URL priority than the dotnet-azure docset (the old docset) the new location should take effect with no OPS changes. The hub that exists currently at https://docs.microsoft.com/dotnet/azure will thus\* be served from this new location with no customer impact.
1. Copied content from Azure/azure-docs-sdk-dotnet.
1. Reorganized content into two broad groups:
    1. General **.NET on Azure** conceptual content (in the root and subdirectories other than *sdk*).
    1. **Azure SDK for .NET** conceptual content (in the *sdk* subdirectory).
1. Removed outdated content and unused includes/media.
1. Created ToCs for both above groups.
1. Created landing page for SDK content.
1. Fixed all build errors (markdown linting, broken links, etc.).
1. Freshened all metadata.
1. Added redirects to account for removed, moved, and renamed content.
1. Breadcrumbs.

### What was NOT done

1. Content that made the cut was not changed beyond tiny incidental changes.
1. The conceptual TOC in Azure/azure-docs-sdk-dotnet will need to change. That will be handled separately after this is merged.
1. The conceptual content in Azure/azure-docs-sdk-dotnet will be archived separately.

*\* -  I think. It's possible I might need SiteHelp to change the priority. But this is my hypothesis.*

### CODE REVIEW

File: docs/azure/TOC.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+- name: Azure for .NET developers
+  href: index.yml
+  items:
+    - name: Introduction to Azure and .NET
+      href: intro.md
+    - name: Migrate to Azure
+      expanded: true
+      items:
+        - name: Choose the right Azure hosting option
```
Comment (scottaddie) [inline docs/azure/TOC.yml]:
```suggestion
        - name: Choose an Azure hosting option
```

**Review by scottaddie — COMMENTED** [review]
I think this PR should also include the necessary *docfx.json* changes. For example, set `fileMetadata` properties, such as the following, if they apply:

- `author`
- `feedback_product_url`
- `feedback_system`
- `ms.author`
- `ms.devlang`
- `ms.prod`
- `ms.technology`

**Review by BillWagner — APPROVED** [review]
Once you address the comments by @scottaddie this is ready to :shipit:

This looks great.

**Comment by CamSoper** [comment]
@scottaddie @BillWagner Would ya'll prefer I put directory-level metadata in docfx.json, or put it all in these files? In the repo they're coming from, I had author, prod, etc. in docfx.json. I would prefer docfx.json, but I'll go with whatever ya'll say.

**Comment by BillWagner** [comment]
@CamSoper We typically put most metadata in docfx.json. Go for it.

**Pull request merged**